src/components/WatchlistCopy/WatchlistCopyPopup.js

```diff
@@ -1,37 +1,215 @@
-import React from 'react'
-import { Checkbox } from '@santiment-network/ui'
-import { Input, Button } from 'semantic-ui-react'
+import React, { useState } from 'react'
+import { connect } from 'react-redux'
+import { compose } from 'recompose'
+import { graphql } from 'react-apollo'
+import Dialog from '@santiment-network/ui/Dialog'
+import Label from '@santiment-network/ui/Label'
+import { ALL_WATCHLISTS_QUERY } from '../../queries/WatchlistGQL'
+import { USER_EDIT_ASSETS_IN_LIST } from '../../actions/types'
+import { sortByAsDates } from '../../utils/sortMethods'
+import { checkIsLoggedIn } from '../../pages/UserSelectors'
+import { showNotification } from '../../actions/rootActions'
+import Watchlists from '../WatchlistPopup/Watchlists'
+import AssetsList from './AssetsList'
+import SearchProjects from '../Search/SearchProjects'
 import styles from './WatchlistCopyPopup.module.scss'
 
 const WatchlistCopyPopup = ({
-  assets,
-  assetsToCopy,
-  onFormSubmit,
-  onChange,
-  onAssetClick
+  assets = [],
+  data = {},
+  trigger,
+  isLoggedIn,
+  watchlistUi: { editableWatchlists },
+  id: currentId,
+  sendChanges,
+  setNotification
 }) => {
+  if (!isLoggedIn) return null
+
+  const [isShown, setIsShown] = useState(false)
+  const [isEditing, setEditing] = useState(false)
+  const [warning, setWarning] = useState(false)
+  const [assetsToCopy, setAssetsToCopy] = useState(new Set())
+  const [watchlistsToCopy, setWatchlistsToCopy] = useState(new Set())
+  const [editWatchlistState, setEditWatchlistState] = useState(
+    editableWatchlists
+  )
+
+  const close = () => {
+    setWatchlistsToCopy(new Set())
+    setAssetsToCopy(new Set())
+    setEditing(false)
+    setIsShown(false)
+  }
+
+  const open = () => setIsShown(true)
+
+  const normalizeListItems = items => items.map(({ project: { id } }) => id)
+
+  const checkRemainingAssets = (listId, assets) => {
+    const list = lists.find(({ id }) => listId === id)
+    const listItems = list ? list.listItems : []
+
+    const remainingAssets = [...assets].filter(id => {
```

Comment:
What is the reason for creating new array? `filter` returns the new array without mutating the old one

Reply:
`assets` has `Set` type